### PR TITLE
Adapt code for modified FreeIPA Terraform module

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request adapts the Terraform code for the changes being made in cisagov/freeipa-server-tf-module#45.

## 💭 Motivation and context ##

We must move the configuration of the Nessus Agent link from [cisagov/ansible-role-cdm-nessus-agent](https://github.com/cisagov/ansible-role-cdm-nessus-agent) to here.  This is because the linking command generates a Tenable Tag, which uniquely identifies the instance to the CDM Nessus server.  If we perform this step at AMI build time, then all instances created from that AMI will have identical Tenable Tags.  Nessus cannot handle "duplicated agents" like this, so we need to instead perform the linking via `cloud-init` at first boot.

See also:
* cisagov/ansible-role-cdm-nessus-agent#11
* cisagov/freeipa-server-tf-module#45
* cisagov/freeipa-server-packer#51

## 🧪 Testing ##

All `pre-commit` hooks pass.  I have also deployed these changes to our staging COOL environment and verified that they function as intended.

## ✅ Pre-approval checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##
* [x] Revert to the `develop` branch of [cisagov/freeipa-server-tf-module](https://github.com/cisagov/freeipa-server-tf-module)

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Excellent! 🏖️

File: freeipa.tf [hunk 1]
```diff
@@ -31,15 +31,19 @@ module "security_groups" {
 
 # Create the IPA servers
 module "ipa0" {
-  source = "github.com/cisagov/freeipa-server-tf-module"
+  source = "github.com/cisagov/freeipa-server-tf-module?ref=improvement%2Flink-nessus-agent"
```
Comment (hillaryj) [inline freeipa.tf]: Should this get stripped back to the original `source` here and lines 56 and 77?

Comment (jsf9k) [reply freeipa.tf]: Yes it should, once cisagov/freeipa-server-tf-module#45 is merged.  There is an item in the pre-merge checklist for this.

Comment (hillaryj) [reply freeipa.tf]: Great! Thanks.

**Review by hillaryj — APPROVED** [review]
🐰

**Pull request merged**